homeassistant/components/minecraft_server/config_flow.py

```diff
@@ -52,9 +52,11 @@ async def async_step_user(self, user_input=None) -> FlowResult:
                     CONF_HOST: host,
                     CONF_PORT: port,
                 }
-                server = MinecraftServer(self.hass, "dummy_unique_id", config_data)
-                await server.async_check_connection()
-                if not server.online:
+                coordinator = MinecraftServerCoordinator(
+                    self.hass, "dummy_unique_id", config_data
+                )
+                server_online = await coordinator.async_is_server_online()
```

Comment:
Please detach the coordinator from the config flow. It is DRY, but it couples your code more. Maybe this can just be a small preliminary PR to remove the couple to MinecraftServer in config flow

Reply:
As I stated in the PR description I would like to finish this PR first and then take care of the optimizations like decoupling the config flow from the coordinator.
The thing is, that currently the connection check is dependent on the SRV check, which I need to remove later, because the API can already handle it. But for this I need changes in the config entry, which will require a new version and thus a new migration function. So you see this is a bigger topic.
If this is not acceptable at all I could think about moving the SRV check in a helper module temporarily so that it can be used also by the config flow directly. Or I could pause this PR and take care of the optimizations first (which will probably make merging later more difficult). Please let me know how I should proceed, thanks.

Reply:
Update: I think I will temporarily move the SRV check to a helper module and decouple config_flow and coordinator in this PR. I did a test and it looks good. I will re-test and push it this evening.